packages/venia-ui/lib/components/Dialog/dialog.css

```diff
@@ -0,0 +1,197 @@
+    opacity: 0.5;
+    visibility: visible;
+    transition-duration: 224ms;
+    transition-timing-function: var(--venia-anim-in);
+}
+
+/* The container is the only grid item in the Dialog grid. */
+.container {
+    background-color: rgb(var(--venia-background-color));
+    border-radius: 5px;
+    box-shadow: 1px 1px 5px rgb(var(--venia-grey-darker));
+
+    /* It sets maximum sizes so its body can handle overflow. */
+    height: 100%;
+    max-height: 90vh;
+    /* Minimum keeps a 16:9 aspect ratio and is 40rem x 22.5rem.  */
+    min-height: 360px;
+    width: 640px;
+
+    /* Nothing is allowed to overflow container itself. */
+    overflow: hidden;
+
+    /* Container is itself a grid container for its children. */
+    display: grid;
+    grid-template-rows: auto 1fr;
+
+    /* Container can be the target of pointer events. */
+    pointer-events: auto;
+}
+
+/*
+ *  Dialog Header styles.
+ */
+
+.header {
+    border-bottom: 1px solid rgb(var(--venia-border));
+    height: 3.5rem;
+    padding: 0 0.75rem 0 1rem;
+
+    /* The Header is itself a grid container for its children. */
+    display: grid;
+    grid-auto-flow: column;
+    grid-template-columns: 1fr auto;
+    column-gap: 1rem;
+    align-items: center;
+}
+
+.headerText {
+    color: rgb(var(--venia-text-alt));
+    text-transform: capitalize;
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
+    line-height: 1.25rem;
+}
+
+.headerButton {
+    /* Horizontally align the close button to the right. */
+    justify-self: right;
+}
+
+/*
+ *  Dialog Body styles.
+ */
+
+.body {
+    padding: 1rem;
+    overflow: auto;
+
+    /* The Body is itself a grid container for its children. */
+    display: grid;
+    grid-template-rows: 1fr auto;
+}
+
+.buttons {
+    padding: 2.5rem;
+
+    display: grid;
+    /* Force the buttons to be the same size (their content could differ). */
+    grid-template-columns: 1fr 1fr;
+    column-gap: 2rem;
+
+    justify-self: center;
+}
+
+.button {
+    overflow: hidden;
+}
+
+/*
+ * Mobile-specific styles.
+ *
+ * Instead of being a full-page modal,
+ * display within the confines of the target drawer.
+ */
+
+@media (max-width: 960px) {
+    .root {
+        left: auto;
+        right: 0;
+        max-width: 360px;
+
+        /* Stretch the content to fill the vertical space. */
+        align-content: stretch;
+
+        /* The Dialog starts off-screen on the right. */
+        transform: translate3d(100%, 0, 0);
+        /* Animate the Dialog's position in mobile too. */
+        transition-property: opacity, transform, visibility;
+    }
+
+    .root_open {
+        composes: root;
+
+        /* The Dialog animates (slides) onto the screen. */
+        transform: translate3d(0, 0, 0);
+    }
+
+    .container {
+        max-height: 100%;
+        width: 100%;
+    }
+
+    .contents {
+        /* The contents should be no wider than the body. */
+        max-width: calc(360px - 2rem);
```

Comment:
I would love to not have to do this but this `contents` div was extending out past its parent `body` div, causing the dialog to clip some content / scroll horizontally.

The `body` class does have `padding` on it but we apply `box-sizing: border-box` to all of our elements so I thought that was supposed to take the padding into account.

At any rate, this is the width of the `body` minus its horizontal `padding`.

Reply:
This has been removed.